api/utils/send_mail.py

```diff
@@ -0,0 +1,30 @@
+# app/email_utils.py
+import smtplib
+from email.mime.text import MIMEText
+from email.mime.multipart import MIMEMultipart
+from api.utils.settings import settings
+
+def send_magic_link(email: str, token: str):
+    '''Sends magic-kink to user email'''
+    sender_email = settings.MAIL_USERNAME
+    receiver_email = email
+    password = settings.MAIL_PASSWORD
+
+    message = MIMEMultipart("alternative")
+    message["Subject"] = "Your Magic Link"
+    message["From"] = sender_email
+    message["To"] = receiver_email
+
+    text = f"Use the following link to log in: http://localhost:8000/magic-link?token={token}"
+    html = f"<html><body><p>Use the following link to log in: <a href='http://localhost:8000/magic-link?token={token}'>Magic Link</a></p></body></html>"

```

Comment:
That hostname should be dynamic
You don't want a user to see local host in their email

So update line 18 andb19